zerver/views/previewable.py

```diff
@@ -0,0 +1,34 @@
+import re
+from urllib.parse import urlparse
+
+from django.http import HttpRequest, HttpResponse
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+from zerver.lib.github import get_issue_or_pr_data
+from zerver.lib.request import REQ, has_request_variables
+from zerver.lib.response import json_success
+from zerver.models import UserProfile
+
+
+@has_request_variables
+def get_previewable_data(
+    request: HttpRequest, user_profile: UserProfile, url: str = REQ()
+) -> HttpResponse:
+    parsed_url = urlparse(url)
+    if (
+        parsed_url.hostname
+        and re.match(r"^(?:www\.)?github\.com$", parsed_url.hostname) is not None
+    ):
+        path_split = parsed_url.path.split("/")
+        if path_split[3] in ["issues", "pull"]:
```

Comment:
If this gets handed `https://github.com` then `parsed_url` is `""`, so `path_split` is `[""]`, which throws a 500 when accessing `path_split[3]`.

Ditto, later, if one were to pass `https://github.com/foo/bar/issues`.